src/ripple/protocol/tests/SecretKey_test.cpp

```diff
@@ -0,0 +1,92 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2012, 2013 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <BeastConfig.h>
+#include <ripple/protocol/PublicKey.h>
+#include <ripple/protocol/SecretKey.h>
+#include <beast/unit_test/suite.h>
+#include <vector>
+
+namespace ripple {
+
+class SecretKey_test : public beast::unit_test::suite
+{
+public:
+    void testDigestSigning()
+    {
+        for (int i = 0; i < 16; i++)
+        {
+            auto const keypair = randomKeyPair (KeyType::secp256k1);
+
+            expect (keypair.first == derivePublicKey (KeyType::secp256k1, keypair.second));
+            expect (*publicKeyType (keypair.first) == KeyType::secp256k1);
+
+            for (int i = 0; i < 256; i++)
+            {
+                uint256 digest;
+                random_fill(digest.data(), digest.size());
+
+                auto const sig = signDigest (
+                    keypair.first, keypair.second, digest);
+
+                expect (sig.size() != 0);
+                expect (verifyDigest (keypair.first,
```

Comment:
a `verifyDigest` that always returns `true` passes this test. We need to check that it correctly returns `false` as well. (same comment for the `testSigning` test below).